lib/src/directory_watcher/windows.dart

```diff
@@ -379,11 +379,11 @@ class _WindowsDirectoryWatcher
   void _startWatch() {
     // Note: "watcher closed" exceptions do not get sent over the stream
     // returned by watch, and must be caught via a zone handler.
-    runZoned(() {
+    runZonedGuarded(() {
       var innerStream = Directory(path).watch(recursive: true);
       _watchSubscription = innerStream.listen(_onEvent,
           onError: _eventsController.addError, onDone: _onDone);
-    }, onError: (error, StackTrace stackTrace) {
+    }, (error, StackTrace stackTrace) {
```

Comment:
The argument types can be inferred now.

https://github.com/dart-lang/watcher/pull/96